vck/src/commonMain/kotlin/at/asitplus/wallet/lib/agent/IssuerAgent.kt

```diff
@@ -209,85 +246,159 @@ class IssuerAgent(
      * returns a JWS representation of that.
      */
-    override suspend fun issueRevocationListCredential(timePeriod: Int?): String? {
+    override suspend fun issueStatusListJwt(time: Instant?) =
+        issueStatusListJwt(time.toTimePeriod())
+            ?: throw IllegalStateException("Status token could not be created.")
+
+    suspend fun issueStatusListJwt(timePeriod: Int?): String? {
+        val tokenPayload = buildStatusListTokenPayload(timePeriod)
+        return wrapStatusListTokenInJws(tokenPayload)
+    }
+
+    /**
+     * Wraps the revocation information from [issuerCredentialStore] into a Status List Token,
+     * returns a CWS representation of that.
+     */
+    override suspend fun issueStatusListCwt(time: Instant?) =
+        issueStatusListCwt(time.toTimePeriod())
+            ?: throw IllegalStateException("Status token could not be created.")
+
+    suspend fun issueStatusListCwt(timePeriod: Int?): ByteArray? {
+        val tokenPayload = buildStatusListTokenPayload(timePeriod)
+        return wrapStatusListTokenInCoseSigned(tokenPayload)
+    }
+
+    /**
+     * Wraps the revocation information from [issuerCredentialStore] into a Token Payload
+     */
+    private fun buildStatusListTokenPayload(timePeriod: Int?): StatusListTokenPayload {
         val revocationListUrl =
             getRevocationListUrlFor(timePeriod ?: timePeriodProvider.getCurrentTimePeriod(clock))
-        val revocationList = buildRevocationList(timePeriod ?: timePeriodProvider.getCurrentTimePeriod(clock))
-            ?: return null
-        val subject = RevocationListSubject("$revocationListUrl#list", revocationList)
-        val credential = VerifiableCredential(
-            id = revocationListUrl,
-            issuer = identifier,
-            issuanceDate = clock.now(),
-            lifetime = revocationListLifetime,
-            credentialSubject = subject
+        val statusList = buildStatusList(timePeriod)
+
+        Napier.d("revocation status list: $statusList")
+        return StatusListTokenPayload(
+            statusList = statusList,
+            issuedAt = clock.now(),
+            subject = UniformResourceIdentifier(revocationListUrl),
         )
-        return wrapVcInJws(credential)
     }
 
     /**
-     * Returns a Base64-encoded, zlib-compressed bitstring of revoked credentials, where
-     * the entry at "revocationListIndex" (of the credential) is true iff it is revoked
+     * Returns a status list, where the entry at "revocationListIndex" (of the credential) is INVALID if it is revoked
      */
-    override fun buildRevocationList(timePeriod: Int?): String? {
-        val bitset = BitSet(REVOCATION_LIST_MIN_SIZE)
-        issuerCredentialStore.getRevokedStatusListIndexList(
+    override fun buildStatusList(timePeriod: Int?): StatusList {
+        return StatusList(
+            buildStatusListView(timePeriod),
+            null,
```

Comment:
that's the aggregationUri ... maybe add this as an constructor parameter to `IssuerAgent`, like the `revocationListBaseUrl` (which may be better named `statusListBaseUrl` now?)

Reply:
fixed in 93ce860696aa4077acb9a053a8b3eac9bee2733b